### PR TITLE
Add interaction listeners for treemap

### PR DESCRIPTION
**Pull request by mcnuttandrew**
This PR addresses the feature enhancement from #174, open to other apis, but I feel like this one matches the the other charts reasonable well while providing access to the full node tree.

### CODE REVIEW

File: src/treemap/index.js [hunk 1]
```diff
@@ -132,10 +135,24 @@ class Treemap extends React.Component {
...
   }
 
+  /**
+   * Triggers a callback on a box if the callback is set.
+   * @param {function} handler Callback function.
+   * @param {Object} d Data point of the arc.
+   * @param {Object} event Event.
+   * @private
+   */
+  _triggerHandler(handler, d, event) {
```
Comment (Apercu) [inline src/treemap/index.js]: Why not setting the default value of the handlers to noop, this way you don't need to do this and the code would be cleaner

File: showcase/treemap/dynamic-treemap.js [hunk 2]
```diff
@@ -19,7 +19,7 @@
 // THE SOFTWARE.
 
 import React from 'react';
-import window from 'global/window';
+import timers from 'timers';
```
Comment (Apercu) [inline showcase/treemap/dynamic-treemap.js]: Didn't knew this one, thanks. Don't it need to be shimmed on the client like `fs`?

File: src/treemap/index.js [hunk 3]
```diff
@@ -132,10 +137,22 @@ class Treemap extends React.Component {
...
   }
 
+  /**
+   * Triggers a callback on a box if the callback is set.
+   * @param {function} handler Callback function.
+   * @param {Object} d Data point of the arc.
+   * @param {Object} event Event.
+   * @private
+   */
+  _triggerHandler(handler = NOOP, d, event) {
```
Comment (Apercu) [inline src/treemap/index.js]: I was more thinking about using the default props so you don't need to create a new method

File: showcase/treemap/dynamic-treemap.js [hunk 4]
```diff
@@ -28,6 +28,7 @@ export default class DynamicTreemapExample extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
+      hoveredNode: false,
```
Comment (Apercu) [inline showcase/treemap/dynamic-treemap.js]: Class prop too?

File: src/treemap/index.js (hunk 3 above)
Comment (mcnuttandrew) [inline src/treemap/index.js]: oh gotcha

File: src/treemap/index.js [hunk 5]
```diff
@@ -20,12 +20,12 @@
...
 import React, {PropTypes} from 'react';
 import * as d3Hierarchy from 'd3-hierarchy';
-import * as d3Color from 'd3-color';
 
 import Animation from 'animation';
 import {CONTINUOUS_COLOR_RANGE, DEFAULT_COLOR, OPACITY_RANGE} from 'theme';
 import {AnimationPropType} from 'utils/animation-utils';
-import {getAttributeFunctor, getMissingScaleProps} from 'utils/scales-utils';
+import {getAttributeFunctor, getFontColorFromBackground, getMissingScaleProps} from 'utils/scales-utils';
+import {} from 'utils/chart-utils';
```
Comment (Apercu) [inline src/treemap/index.js]: ?

**Pull request merged**